modules/caddyhttp/caddyhttp.go

```diff
@@ -255,17 +255,17 @@ func CleanPath(p string, collapseSlashes bool) string {
 
 	// replace repeated slashes with an invalid/impossible URI character
 	// to avoid collisions; then clean the path, then replace them back
-	const tmpSlash = 0xff
+	var tmpSlash = string("\xff/")
 	var sb strings.Builder
 	for i, ch := range p {
 		if ch == '/' && i > 0 && p[i-1] == '/' {
-			sb.WriteByte(tmpSlash)
+			sb.WriteString(tmpSlash)
 			continue
 		}
 		sb.WriteRune(ch)
 	}
 	halfCleaned := cleanPath(sb.String())
-	halfCleaned = strings.ReplaceAll(halfCleaned, string([]byte{tmpSlash}), "/")
+	halfCleaned = strings.ReplaceAll(halfCleaned, "/\xff", "/")
```

Comment:
This fix looks good to me!

Another solution I can think of - with minimum changes - is to insert `0xff` into two consecutive slashes. For example:

```diff
	const tmpSlash = 0xff
	var sb strings.Builder
	for i, ch := range p {
		if ch == '/' && i > 0 && p[i-1] == '/' {
			sb.WriteByte(tmpSlash)
-			continue
		}
		sb.WriteRune(ch)
	}
	halfCleaned := cleanPath(sb.String())
-	halfCleaned = strings.ReplaceAll(halfCleaned, string([]byte{tmpSlash}), "/")
+	halfCleaned = strings.ReplaceAll(halfCleaned, string([]byte{tmpSlash}), "")
```

That being said, the two solutions are essentially equivalent, so just pick the one you prefer :)

Reply:
@RussellLuo That's a way better solution IMO! Let's do that. Do you want to push a commit to this branch or something? That way you can get credit for it. (I just gave you write permissions. Welcome to the team!) If you don't care then I don't mind doing it. Either way.

Reply:
Oh, thanks! It's my pleasure 😄

Reply:
I have pushed a commit (also improved the docs). Hope it's ok :)

Reply:
Yes! Looks good. Thank you very much!